mitodl/hugo-course-publisher
# build course downloads

#### Pre-Flight checklist


#### What are the relevant tickets?
https://trello.com/c/bVQwTyeU/170-download-course-material-should-be-a-button-in-course-info

#### What's this PR do?
This PR adds the `build:zips` command to `package.json`.  Running this script will first perform the pdfjs and webpack builds, then crawl the Hugo content tree and generate a static site using the `hugo` command for each course.  These sites are individually packaged into their own respective zip files and placed in a folder called `zips` inside the Hugo `static` folder.  This way, if they are built and put in place they are deployed with the site and are always available at an absolute url from the site root with the path: `/zips/{{ .Params.course_id }}.zip`.

#### How should this be manually tested?
Run `npm run build:zips` to produce the zip archives.  Then, run the multi-course site with `npm start`.  "Download Course Materials" buttons should provide you with a zip archive that contains a browsable static site with the root being the course you downloaded it from.  Please note that if you generate the zips and then switch branches, you will likely need to remove the `site/static/zips` folder.  This PR contains the change to `.gitignore` to make sure they are ignored.
